scipy/fft/czt.py

```diff
@@ -0,0 +1,548 @@
+# This program is public domain
+# Authors: Paul Kienzle, Nadav Horesh
+"""
+Chirp z-transform.
+
+We provide two interfaces to the chirp z-transform: an object interface
+which precalculates part of the transform and can be applied efficiently
+to many different data sets, and a functional interface which is applied
+only to the given data set.
+
+Transforms
+----------
+
+CZT : callable (x, axis=-1) -> array
+   Define a chirp z-transform that can be applied to different signals.
+ZoomFFT : callable (x, axis=-1) -> array
+   Define a Fourier transform on a range of frequencies.
+
+Functions
+---------
+
+czt : array
+   Compute the chirp z-transform for a signal.
+zoomfft : array
+   Compute the Fourier transform on a range of frequencies.
+"""
+
+import cmath
+import numpy as np
+from numpy import pi, arange
+from scipy.fft import fft, ifft, next_fast_len
+
+__all__ = ['czt', 'zoomfft', 'CZT', 'ZoomFFT', 'czt_points']
+
+
+def _validate_sizes(n, m):
+    if n < 1:
+        raise ValueError("Invalid number of CZT data "
+                         "points (%d) specified." % n)
+
+    if int(n) != n:
+        raise ValueError('n must be a positive integer')
```

Comment:
That would raise an exception when someone tries something like `n = fs/4`, though, even though the argument has a valid integer value

Reply:
At some point in the last couple of years there was a lengthy discussion in SciPy of how to check for int-ness. The operator method was chosen. I think we should stick to that convention rather than deviate and/or debate here.

Reply:
IMO that's not a downside. Casting `float` to `int` is sensitive to changes at the machine epsilon range. So, user code might randomly fail the test for some values but not others. So strict typing of integers is preferable IMO.